java/maven.hints/src/org/netbeans/modules/maven/hints/pom/ParentVersionError.java

```diff
@@ -107,7 +108,17 @@ public List<ErrorDescription> getErrorsForDocument(POMModel model, Project prj)
                     if (parentPrj != null) {
                         NbMavenProject nbprj = parentPrj.getLookup().lookup(NbMavenProject.class);
                         if (nbprj != null) { //do we have some non-maven project maybe?
-                            MavenProject mav = nbprj.getMavenProject();
+                            MavenProject mav = nbprj.getMavenProject();  
+                            if(PomModelUtils.isPropertyExpression(declaredVersion)){
+                                String key = PomModelUtils.getPropertyName(declaredVersion);
+                                String propVal = PomModelUtils.getProperty(model, declaredVersion);
+                                if(propVal != null){
+                                    declaredVersion = propVal;
+                                }
+                                else {
+                                    declaredVersion = mav.getProperties().getProperty(key, declaredVersion);
+                                }
+                            }
```

Comment:
i think what you want to do is this:
```java
                            MavenProject mav = nbprj.getMavenProject();
                            if (PomModelUtils.isPropertyExpression(declaredVersion)) {
                                String propVal = PomModelUtils.getProperty(model, declaredVersion);
                                if (propVal != null) {
                                    declaredVersion = propVal;
                                }
                            }
```
since `PomModelUtils.getProperty` is already doing what you put into the else branch.

I did also format it using NB default formatting. e.g space after `if` etc - feel free to paste it if its what you intended.

Reply:
I want to get declaredVersion value first from the present pom file and if it is not present there then from the parent pom file i.e. from the mav but when I tried `mav.getModel()` it is of incompatible data type with PomModelUtils that's why I have written in the following manner.

Reply:
oh wait. that is the wrong `POMmodel` - let me think about this a bit

Reply:
```java
                            MavenProject mav = nbprj.getMavenProject();
                            if (PomModelUtils.isPropertyExpression(declaredVersion)) {
                                String propVal = PomModelUtils.getProperty(model, declaredVersion);
                                if (propVal != null) {
                                    declaredVersion = propVal;
                                } else {
                                    String key = PomModelUtils.getPropertyName(declaredVersion);
                                    declaredVersion = mav.getProperties().getProperty(key, declaredVersion);
                                }
                            }
```
I think you are right, I did only format it and moved the key into the else branch. I am not 100% sure if the order is correct, e.g if it should check the parent first and the current pom second or if this is correct - but it makes sense.

Reply:
I checked and it seems maven first searches for the property value in the child POM and if it is not present there then it searches in the parent POM file. Thanks for formatting the code.

Reply:
thanks for checking, yes the override behavior makes sense.